### PR TITLE
BREAKING: convert src and tests to LitElement

### PR DESCRIPTION
**Pull request by web-padawan**
Extracted from #143 in order to make it easier to review.
This PR only contains changes to `src` and unit tests. 

The tests that have to be moved to visual tests ( equal width and flex tabs) are not included.

Base branch is set to `next-prepare` so we can merge from there to `next` later with 1 commit.

### CODE REVIEW

File: src/vaadin-tabs-mixin.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      super._sizeChanged && super._sizeChanged(contentRect);
+    }
+
+    private _onScroll() {
+      this._updateOverflow();
+    }
+
+    private get _overflow() {
+      const { scrollTop, scrollLeft, scrollHeight, scrollWidth } = this._scrollTarget;
+      const scrollPosition = this._vertical ? scrollTop : scrollLeft;
```
Comment (yuriy-fix) [inline src/vaadin-tabs-mixin.ts]: Question: if we are going to use `isVertical` from `KeyboardDirection` and we anyway need to override it in tabs. Can we at the same time get rid out of `_vertical` in `SelectionInView` and use `isVertical` from `KeyboardDirection` over there and here as well? 🤔

File: src/vaadin-tabs.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import { customElement } from 'lit-element';
+import { TabsBase } from './vaadin-tabs-base';
+import { TabsMixin } from './vaadin-tabs-mixin';
+import './vaadin-tab';
+
+/**
+ * `<vaadin-tabs>` is a Web Component for easy switching between different views.
+ *
+ * @attr {start | none | both} overflow - Attribute set depending on whether the items fit into viewport.
```
Comment (yuriy-fix) [inline src/vaadin-tabs.ts]: Should we mention `@attr {horizontal | vertical} orientation` as well?

File: src/vaadin-tab-mixin.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    protected _onKeyUp(event: KeyboardEvent) {
+      const willClick = this.hasAttribute('active');
+
+      super._onKeyUp && super._onKeyUp(event);
+
+      if (willClick) {
+        const anchor = this.querySelector('a');
+        if (anchor) {
+          anchor.click();
```
Comment (hongduc-phan) [inline src/vaadin-tab-mixin.ts]: Question: `anchor` seems always false and it does not implement anchor.click() ?

Comment (hongduc-phan) [reply src/vaadin-tab-mixin.ts]: understand after asking. thanks.

**Comment by hongduc-phan** [comment]
when the tab has display none, the tab still has tabIndex = 0, and can not move left-right to choose other tabs

File: src/vaadin-tabs-mixin.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+        this._updateOverflow();
+      }
+
+      // Ensure scroll to item is called after "overflow" update
+      super.updated(props);
+    }
+
+    protected _sizeChanged(contentRect: DOMRect) {
+      this._updateOverflow();
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply src/vaadin-tabs-mixin.ts]: This is for convenience, to make sure we do it before `resize` event is fired.

I will add a comment to clarify this.

File: package.json [hunk 5]
```diff
@@ -18,30 +18,68 @@
...
+    "test:debug": "karma start --single-run false --browsers Firefox",
+    "test:sauce": "cross-env TEST_PLATFORM=sauce karma start",
+    "test:visual": "npm run build && gemini test test/visual"
+  },
+  "dependencies": {
+    "@vaadin/element-base": "^0.1.3",
+    "@vaadin/active-state-mixin": "^0.1.1",
+    "@vaadin/disabled-state-mixin": "^0.1.0",
+    "@vaadin/focus-visible-mixin": "^0.1.1",
+    "@vaadin/keyboard-direction-mixin": "^0.1.2",
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply package.json]: Done.

File: src/vaadin-tabs-mixin.ts [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+      Promise.all(tabsComplete).then(() => {
+        this._updateOverflow();
+      });
+    }
+
+    protected updated(props: PropertyValues) {
+      if (props.has('orientation') || props.has('_items')) {
+        this.items.forEach(item => {
+          if (this.orientation) {
+            item.setAttribute('orientation', this.orientation);
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply src/vaadin-tabs-mixin.ts]: Done.

File: src/vaadin-tabs-mixin.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (web-padawan) [reply src/vaadin-tabs-mixin.ts]: Added the comment.

File: src/vaadin-tabs-mixin.ts (hunk 1 above)
Comment (web-padawan) [reply src/vaadin-tabs-mixin.ts]: Now we have `_vertical` used in all mixins, so no need to change it here.

File: src/vaadin-tabs.ts (hunk 2 above)
Comment (web-padawan) [reply src/vaadin-tabs.ts]: Currently the `orientation` property is picked up from `OrientationMixin`.

The `@attr` is currently used for state attributes (that element sets on itself).
So I don't think we need to list `orientation` here for now.

File: package.json [hunk 7]
```diff
@@ -2,14 +2,14 @@
   "name": "@vaadin/vaadin-tabs",
   "version": "3.0.5",
   "description": "vaadin-tabs",
-  "main": "vaadin-tabs.html",
+  "main": "vaadin-tabs.js",
```
[the remark this answers is not in the document]
Comment (web-padawan) [reply package.json]: Done.

**Comment by web-padawan** [comment]
> when the tab has display none, the tab still has tabIndex = 0

Created https://github.com/vaadin/component-mixins/issues/34 to address that.

**Pull request merged**